### PR TITLE
WIP: list resource in form request using parent key instead using pks

### PR DESCRIPTION
**Pull request by petrus-v**
This PR is intended to works with https://github.com/AnyBlok/anyblok_furetui/pull/13

- [x] remove pks mechanism and replace by a request using remote_columns request
- [x] manage new entries
- [x] manage update entries
- [x] manage delete entries (it's removed in backend on saving, the DELETE_DATA (data store commit) do not properly remove the line or something get the line back) so user still see the removed line
- [x] at the moment saving the x2m field form resource update the backend without waiting the save on the main page. I don't think it's what we expected
- [x] only last change on the o2m is send in the patch while saving (fixed by 29aa20904390deb60bf3e04722cd465511b5bbda)
- [x] replace ``uuid`` by ``__x2m_uuid`` as previously set (addressed by 412cdd61cdc3b1d76887f7ffc8d691eabe54f481)
- [x] Cancel button remove news (is there a clrear somewhere ?) (addressed by 412cdd61cdc3b1d76887f7ffc8d691eabe54f481)
- [x] When saving (in the main form resource) the o2m HTTP request to the server is not launched. I think it would be a good idea to get fresh data and clear local changes,
- [x] rename Cancel button => Reset to initial values | Save => Apply
- [x] manage pagination and counter with added/deleted
  * [x] add unittest
- [x] display counters (total, add, deleted, updated)
  * [ ] add unittest
- [x] implement revert method
- [x] deleted lines shoould not be clickable => to review 29d7f9af152a0f55919630da789d5a5e4c514e73
- [x] save after modification must set the form view in readonly mode
- [ ] ~only one __x2m_state change in one2many change~ => issue #43 
- [ ] ~allow developer to manage new record position (top/end/sequence)~ => issue #44 
- [ ] ~manage delete new / update and delete / delete and update / new and update~ => issue #43 
- [ ] ~manage tab counter (cf https://github.com/AnyBlok/furet_ui/pull/41#issuecomment-612090398)~ => issue #42 

---
Todo later:
- [ ] ~when adding a new row in o2m where there is an m2o set to null the first time we create it, if we set it the second time m2o field is not set. Otherwise m2o field is properly changing if set the first time.~ => issue   #45

### CODE REVIEW

**Comment by petrus-v** [comment]
About the counter, previously threre is an attribute on the BTabItem:
``$attrs.counter:"fields.items.length"``

Which is used by the ``furet-ui-tab`` component which use the `eval_counter` where
```javascript
  try {
    res = eval(value);
    if (typeof res === 'object') {
      if (res.length !== undefined) res = res.length
      else if (res.count !== undefined) res = res.count
    } else if (res == undefined) res = 0;
  } catch (e) {
    console.log(e)
  }
```
 
AnyBlok/anyblok_furetui#13 stopped to returns ``items`` fields. 

That why I'm getting the following error:
```javascript
TypeError: Cannot read property 'length' of undefined
    at eval (eval at eval_counter (helper.js?e270:117), <anonymous>:1:14)
    at eval_counter (helper.js?e270:67)
    at render (helper.js?e270:90)
    at createFunctionalComponent (vue.esm.js?a026:3067)
    at createComponent (vue.esm.js?a026:3240)
    at _createElement (vue.esm.js?a026:3431)
    at createElement (vue.esm.js?a026:3362)
    at Proxy.vm._c (vue.esm.js?a026:3500)
    at Proxy.eval (eval at createFunction (vue.esm.js:NaN), <anonymous>:3:1305)
    at VueComponent.Vue._render (vue.esm.js?a026:3557)
```

File: src/components/fields/relationship/one2many.js [hunk 1]
```diff
@@ -110,33 +107,26 @@ defineComponent('furet-ui-form-field-one2many', {
...
+        this.updateValue(this.addState(actions, "UPDATED"), actions.changes);
       },
-      o2m_delete (actions) {
-        this.updateValue(this.addState(actions, 'DELETED'))
-      },
-    },
+      o2m_delete(actions) {
+        this.updateValue(this.addState(actions, "DELETED"));
+        //TODO: figure out why this commit is not taken in consideration
+        this.$store.commit("DELETE_DATA", actions);
```
Comment (petrus-v) [inline src/components/fields/relationship/one2many.js]: I couldn't figure out why this was not updating the state of the store properly.

Comment (petrus-v) [reply src/components/fields/relationship/one2many.js]: In fact I understood...
The commit is properly done, the matter is that when the user come back to the o2m list (delete button is on the form resource) a call to the server is made but we haven't ask the server to save that change yet.

I think we shouldn't call the server until the main save.

Or consider this component independent and save changes in x2m in a different transaction.... which is not what we wants !

**Comment by jssuzanne** [comment]
> About the counter, previously threre is an attribute on the BTabItem:
> `$attrs.counter:"fields.items.length"`
> 
> Which is used by the `furet-ui-tab` component which use the `eval_counter` where
> 
> ```js
>   try {
>     res = eval(value);
>     if (typeof res === 'object') {
>       if (res.length !== undefined) res = res.length
>       else if (res.count !== undefined) res = res.count
>     } else if (res == undefined) res = 0;
>   } catch (e) {
>     console.log(e)
>   }
> ```
> 
> [AnyBlok/anyblok_furetui#13](https://github.com/AnyBlok/anyblok_furetui/pull/13) stopped to returns `items` fields.
> 
> That why I'm getting the following error:
> 
> ```js
> TypeError: Cannot read property 'length' of undefined
>     at eval (eval at eval_counter (helper.js?e270:117), <anonymous>:1:14)
>     at eval_counter (helper.js?e270:67)
>     at render (helper.js?e270:90)
>     at createFunctionalComponent (vue.esm.js?a026:3067)
>     at createComponent (vue.esm.js?a026:3240)
>     at _createElement (vue.esm.js?a026:3431)
>     at createElement (vue.esm.js?a026:3362)
>     at Proxy.vm._c (vue.esm.js?a026:3500)
>     at Proxy.eval (eval at createFunction (vue.esm.js:NaN), <anonymous>:3:1305)
>     at VueComponent.Vue._render (vue.esm.js?a026:3557)
> ```



> About the counter, previously threre is an attribute on the BTabItem:
> `$attrs.counter:"fields.items.length"`
> 
> Which is used by the `furet-ui-tab` component which use the `eval_counter` where
> 
> ```js
>   try {
>     res = eval(value);
>     if (typeof res === 'object') {
>       if (res.length !== undefined) res = res.length
>       else if (res.count !== undefined) res = res.count
>     } else if (res == undefined) res = 0;
>   } catch (e) {
>     console.log(e)
>   }
> ```
> 
> [AnyBlok/anyblok_furetui#13](https://github.com/AnyBlok/anyblok_furetui/pull/13) stopped to returns `items` fields.
> 
> That why I'm getting the following error:
> 
> ```js
> TypeError: Cannot read property 'length' of undefined
>     at eval (eval at eval_counter (helper.js?e270:117), <anonymous>:1:14)
>     at eval_counter (helper.js?e270:67)
>     at render (helper.js?e270:90)
>     at createFunctionalComponent (vue.esm.js?a026:3067)
>     at createComponent (vue.esm.js?a026:3240)
>     at _createElement (vue.esm.js?a026:3431)
>     at createElement (vue.esm.js?a026:3362)
>     at Proxy.vm._c (vue.esm.js?a026:3500)
>     at Proxy.eval (eval at createFunction (vue.esm.js:NaN), <anonymous>:3:1305)
>     at VueComponent.Vue._render (vue.esm.js?a026:3557)
> ```

I merges the private project where the length is removed from the template of the product

File: src/components/fields/relationship/one2many.js (hunk 1 above)
Comment (jssuzanne) [reply src/components/fields/relationship/one2many.js]: Yes it is the reason why we saved the list of primary keys

**Pull request merged**